pkg/policy/k8s/watcher.go

```diff
@@ -49,6 +55,124 @@ type policyWatcher struct {
 	// cidrGroupPolicies is the set of policies that contain ToServices references
 	toServicesPolicies map[resource.Key]struct{}
 	cnpByServiceID     map[k8s.ServiceID]map[resource.Key]struct{}
+	// maps cnp file name to cnp object. this is required to retrieve data during delete.
+	fileNameToCnpCache map[string]*types.SlimCNP
+}
+
+func (p *policyWatcher) translateToCNPObject(file string) (*cilium_v2.CiliumNetworkPolicy, error) {
+	data, err := os.ReadFile(file)
+	if err != nil {
+		return nil, err
+	}
+	jsonData, err := yaml.YAMLToJSON([]byte(data))
+	if err != nil {
+		return nil, err
+	}
+
+	cnp := &cilium_v2.CiliumNetworkPolicy{}
+	err = json.Unmarshal(jsonData, cnp)
```

Comment:
I also think we should also think about what we want to do with the CNP fields which don't make sense when loaded from file, e.g. K8s name and namespace. In particular if there are two files with the same policy name on the object metadata, then I think the current system is non-deterministic in what policy gets applied.

Instead of parsing in the full `cilium_v2.CiliumNetworkPolicy` schema, we could consider only reading in `api.Rule` and overwrite the labels of each rule to contain the filename or something.

Reply:
If there are 2 policies with same name, then it would merge both. If policy is same, then it would ignore adding same rule and if different it merges rules from both policy. Wouldn't namespace be significant if user wants to apply policy for specific namespace.? If we skip name and namespace, then it supports only clusterwide policies which is enough to solve our usecase. I'm fine with that as well.

Reply:
yep correct. instead of name, can update with filename instead but still I like to namespace. wdyt?

Reply:
Ah, I only saw this after I re-reviewed. Yes, I think keeping the namespace and forcing the name to be the filename seems like an elegant solution